### DISCUSSION TITLE
How to include CEF with H.264 into CEFSharp nuget?

### DISCUSSION THREAD

**Discussion by bryanforst** [cefsharp/CefSharp#3731] [Q&A]
I have built and tested a CEF build with h.264 operating properly. 
Trying to now get this files into CEFSharp nuget packages for WPF 64 bit

I thought I could build a CEFSharp nuget package by:
1. run build.ps1 vs2019 to get build files completely created. 
 
2. Replace files in packages\cef.redist.x64.91.1.23\CEF folder with files in 
    chromium\src\out\Release_GN_x64 folder using 
    https://github.com/cefsharp/CefSharp/wiki/Output-files-description-table-%28Redistribution%29
    as a reference

3. run build.ps1 nupkg-only

4. Take nuget files from NuGet folder and use them locally in my VS C# app

Whenever I do this I see the standard CEF files in my app's bin and the nuget packages folder for cef redistribution. 

I have also tried to directly copy the CEFSharp packages\cef.redist.x64.91.1.23
into  my app packages/cef.redist.x64.91.1.23 but the build still puts the wrong files into my bin folder

All help is appreciated.

**Comment by bryanforst** [comment]
Additional info:

After running  build.ps1 vs2019 I then copied the files from my CEF/h.264  release area into the packages/cef.redist.x64.91.1.23\CEF folder.

Then I copied the .nuspec file from the cef.redist.x64.91.1.23.nupkg file in the folder and modified it as an h264 variant. 
Then I build a new cef.redist.x64.91.1.23.nupkg file from my .nuspec
And then re-ran build.ps1 vs2019 noticing that it picked up my cef.redist.x64.91.1.23.nupkg file and did not redeploy it from the nuget repo

Using the resulting files locally: cef.redist.x64.91.1.23.nupkg, CefSharp.Common.91.1.230.nupkg, and CefSharp.Wpf.91.1.230.nupkg
I rebuilt my app. 

It still fails the h264 test in html5test.com !?!

Im only modifying the 64bit cef.redist because my app configuration is for 64 bit. 
Is there a chance the cef 32bit version is being used????? 

Any other pieces of advice?

**Reply by amaitland** [reply]
> Is there a chance the cef 32bit version is being used?????

If your PlatformTarget is AnyCpu and you have Prefer32Bit set to true then you will have a 32bit exe which will use the 32bit dlls.

This is the default for many of the visual studio templates. Make sure you are targeting x64.

https://stackoverflow.com/a/12066861/4583726

**Reply by bryanforst** [reply]
yup, thats my target x64

**Reply by amaitland** [reply]
Double check your PlatformTarget is x64 as it's possible your solution target might differ from the PlatformTarget. 

https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.WinForms/CefSharp.MinimalExample.WinForms.csproj#L34

Confirm in task manager your process is x64.

Past that unfortunately it's not clear what's going on here in terms of file copies.

**Reply by bryanforst** [reply]
hmm, she is definitely running at 64 bit. 
I will rerun the entire process to make sure I have not missed a step.

**Reply by amaitland** [reply]
Make sure you delete the original packages from the Nuget cache, packages folder and bin folder is empty.

**Reply by bryanforst** [reply]
Ive cleaned out the various caches and folders and rebuilt from scratch, same issue - no h264 support. 

Verified my CEF build has h264 support. Again copied files from 
chromium\src\out\Release_GN_x64 to packages\cef.redist.x64.91.1.23\CEF 
and recreated the cef.redist.x64.91.1.23.nupkg file before running build.ps1 vs2019 one more time

What Im not clear on is how files from packages\cef.redist.x64.91.1.23\CEF or the cef.redist.x64.91.1.23.nupkg
i created is used to create the Common and WPF nuget files. 
Im assuming  its necessary to install cef.redist.x64.91.1.23.nupkg that i created and not the one from nuget.org

**Reply by bryanforst** [reply]
I prefer to create a nupkg delivery but I think I will try adding the cefsharp files directly to the app and see if at least that works. We had a fellow here about 1.5 years ago who created a local nuget version with h264 embedded but unfortunately he did not document the process. 

Is there a page outlining these steps? Otherwise I will look through the discussions and  see what i can find

**Reply by bryanforst** [reply]
OK, some changes. 
Looking at Debug/Options menu under Nuget i clicked on "clear all from Nuget cache". 
That seemed to unstick me. I had to update the installs on a few items but at least its deploying correctlly. 
Getting some startup errors about the .pak files not being correct - will check those out as I think i saw someone else
commenting on that
